### PR TITLE
Upgrade Azure.Identity and MSAL versions and WAM

### PR DESCRIPTION
**Pull request by msJinLei**
## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Accounts/Accounts/ChangeLog.md [hunk 1]
```diff
@@ -19,6 +19,9 @@
 -->
 
 ## Upcoming Release
+* Enable WAM as default way of interactive login flow.
```
Comment (isra-fel) [inline src/Accounts/Accounts/ChangeLog.md:22]: We might want to be more informative about WAM being default (for example, add how to switch back to the legacy experience) and less informative about package upgrade (because of the length limitation of psd1)

File: src/Accounts/AssemblyLoading/ConditionalAssemblyProvider.cs [hunk 2]
```diff
@@ -42,15 +42,16 @@ public static void Initialize(string rootPath, IConditionalAssemblyContext conte
...
                 CreateAssembly("netstandard2.0", "Microsoft.Bcl.AsyncInterfaces", "1.0.0.0"),
-                CreateAssembly("netstandard2.0", "Microsoft.Identity.Client", "4.56.0.0"),
-                CreateAssembly("netstandard2.0", "Microsoft.Identity.Client.Extensions.Msal", "4.56.0.0"),
-                CreateAssembly("netstandard2.0", "Microsoft.Identity.Client.Broker", "4.56.0.0"),
+                CreateAssembly("netstandard2.0", "Microsoft.Identity.Client", "4.60.3.0"),
+                CreateAssembly("netstandard2.0", "Microsoft.Identity.Client.Extensions.Msal", "4.60.3.0"),
+                CreateAssembly("netstandard2.0", "Microsoft.Identity.Client.Broker", "4.60.3.0"),
                 CreateAssembly("netstandard2.0", "Microsoft.Identity.Client.NativeInterop", "0.13.8.0"),
                 CreateAssembly("netstandard2.0", "Microsoft.IdentityModel.Abstractions", "6.22.1.0"),
+                CreateAssembly("netstandard2.0", "System.ClientModel", "1.0.0.0"),
```
Comment (isra-fel) [inline src/Accounts/AssemblyLoading/ConditionalAssemblyProvider.cs:54]: Who depends on this?

File: src/Accounts/Authentication/Utilities/AzConfigReader.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
+﻿using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using Microsoft.Azure.Commands.Shared.Config;
+using Microsoft.Azure.PowerShell.Common.Config;
+
+namespace Microsoft.Azure.Commands.Common.Authentication.Utilities
+{
+    static public class AzConfigReader
```
Comment (isra-fel) [inline src/Accounts/Authentication/Utilities/AzConfigReader.cs:7]: make it internal & add licence header

File: src/Accounts/Authentication/Utilities/AzConfigReader.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        {
+            get
+            {
+                return AzureSession.Instance;
+            }
+        }
+
+        static public bool IsWamEnabled()
+        {
+            if (!Session.TryGetComponent<IConfigManager>(nameof(IConfigManager), out var config))
```
Comment (isra-fel) [inline src/Accounts/Authentication/Utilities/AzConfigReader.cs:20]: Session can be null?

File: src/Network/Network/help/New-AzVirtualNetworkGatewayPolicyGroupMember.md
Comment (isra-fel) [inline src/Network/Network/help/New-AzVirtualNetworkGatewayPolicyGroupMember.md:1]: Why is this file included in the PR?

File: tools/StaticAnalysis/HelpAnalyzer/HelpAnalyzer.cs [hunk 5]
```diff
@@ -287,6 +287,8 @@ public void Analyze(IEnumerable<string> scopes, IEnumerable<string> modulesToAna
             {
                 if (!cmdletDict.ContainsKey(helpRecord))
                 {
+                    Console.WriteLine($"ModuleName = {moduleName}, Cmdlet Count = {cmdlets.Count}");
```
Comment (isra-fel) [inline tools/StaticAnalysis/HelpAnalyzer/HelpAnalyzer.cs:290]: Wanna keep this?

**Review by isra-fel — CHANGES_REQUESTED** [review]
Default value of the `EnableLoginByWam` config should be updated, its description as well.

File: src/Accounts/AssemblyLoading/ConditionalAssemblyProvider.cs (hunk 2 above)
Comment (msJinLei) [reply src/Accounts/AssemblyLoading/ConditionalAssemblyProvider.cs:54]: introduced by Azure.Core 1.38.0

File: tools/StaticAnalysis/HelpAnalyzer/HelpAnalyzer.cs (hunk 5 above)
Comment (msJinLei) [reply tools/StaticAnalysis/HelpAnalyzer/HelpAnalyzer.cs]: Removed

File: src/Network/Network/help/New-AzVirtualNetworkGatewayPolicyGroupMember.md
Comment (msJinLei) [reply src/Network/Network/help/New-AzVirtualNetworkGatewayPolicyGroupMember.md:1]: To fix a static analysis issue

File: src/Accounts/Authentication/Utilities/AzConfigReader.cs (hunk 4 above)
Comment (msJinLei) [reply src/Accounts/Authentication/Utilities/AzConfigReader.cs]: AzureSession is initialized when Az.Accounts is imported and so is not expected to be null. If null an exception is expected.

File: src/Accounts/Authentication/Utilities/AzConfigReader.cs (hunk 3 above)
Comment (msJinLei) [reply src/Accounts/Authentication/Utilities/AzConfigReader.cs]: Done

File: src/Accounts/Authentication/Utilities/AzConfigReader.cs [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using Microsoft.Azure.Commands.Shared.Config;
+using Microsoft.Azure.PowerShell.Common.Config;
+
+namespace Microsoft.Azure.Commands.Common.Authentication.Utilities
+{
+    static public class AzConfigReader
+    {
+
+        public static IAzureSession Session
```
Comment (BethanyZhou) [inline src/Accounts/Authentication/Utilities/AzConfigReader.cs:24]:
```suggestion
        internal static IAzureSession Session
```
only expose public methods

File: .azure-pipelines/security-tools.yml [hunk 7]
```diff
@@ -60,14 +60,21 @@ jobs:
...
 
   - task: securedevelopmentteam.vss-secure-development-tools.build-task-binskim.BinSkim@3
     displayName: Run BinSkim
     inputs:
-      InputType: 'Basic'
-      Function: 'analyze'
-      AnalyzeTarget: '$(Build.SourcesDirectory)/artifacts/Debug/*.dll;$(Build.SourcesDirectory)/artifacts/*.exe'
-      AnalyzeStatistics: true
+      InputType: 'CommandLine'
+      arguments: 'analyze @$(Build.SourcesDirectory)/artifacts/MyFileList.rsp --recurse'
```
Comment (BethanyZhou) [inline .azure-pipelines/security-tools.yml:77]: Verified by https://github.com/Azure/azure-powershell/pull/24902

File: src/Accounts/Authentication/Identity/MsalConfidentialClient.cs [hunk 8]
```diff
@@ -280,7 +292,11 @@ protected virtual async ValueTask<IConfidentialClientApplication> CreateClientCo
 
             if (!string.IsNullOrEmpty(tenantId))
             {
-                builder.WithAuthority(AuthorityHost.AbsoluteUri, tenantId);
+                var uriBuilder = new UriBuilder(AuthorityHost.AbsoluteUri)
+                {
+                    Path = tenantId
+                };
+                builder.WithTenantIdFromAuthority(uriBuilder.Uri);
```
Comment (isra-fel) [inline src/Accounts/Authentication/Identity/MsalConfidentialClient.cs:299]: Refactor to avoid duplication (4 times this pattern appears in the source code)

File: src/Accounts/Authentication/Properties/Resources.resx [hunk 9]
```diff
@@ -381,7 +381,7 @@
     <comment>0 = key of the config</comment>
   </data>
   <data name="HelpMessageOfEnableWamLogin" xml:space="preserve">
-    <value>[Preview] When enabled, Web Account Manager (WAM) will be the default interactive login experience. It will fall back to using the browser if the platform does not support WAM. Note that this feature is under preview. Microsoft Account (MSA) is currently not supported. Feel free to reach out to Azure PowerShell team if you have any feedbacks: https://aka.ms/azpsissue</value>
+    <value>When enabled, Web Account Manager (WAM) will be the default interactive login experience. It will fall back to using the browser if the platform does not support WAM. Feel free to reach out to Azure PowerShell team if you have any feedbacks: https://aka.ms/azpsissue</value>
```
Comment (isra-fel) [inline src/Accounts/Authentication/Properties/Resources.resx:384]: Refer to the document

File: src/Accounts/Authentication/Utilities/AzConfigReader.cs (hunk 3 above)
Comment (isra-fel) [reply src/Accounts/Authentication/Utilities/AzConfigReader.cs:23]: I still have concerns about the design of AzConfigReader. For example, under what circumstances should you add new method to this type? Whenever a new config is introduced? But we can discuss it later since changing its design won't be breaking change

File: src/Accounts/Authentication/Factories/AuthenticationFactory.cs [hunk 10]
```diff
@@ -184,28 +185,23 @@ private static AzPSAuthenticationFailedException AnalyzeMsalException(Exception
...
-                        {
-                            errorMessage = $"You must use multi-factor authentication to access resource {resourceId}, please rerun 'Connect-AzAccount' with additional parameter '-AuthScope {resourceId}'.";
-                            desensitizedMessage = "MFA is required to access resource";
-                        }
-                        return new AzPSAuthenticationFailedException(
-                            errorMessage,
-                            msalUiRequiredException.ErrorCode,
-                            originalException,
-                            desensitizedMessage: desensitizedMessage);
+                        errorMessage = $"You must use multi-factor authentication to access tenant {tenantId}, please rerun 'Connect-AzAccount' with additional parameter '-TenantId {tenantId}'.";
```
Comment (isra-fel) [inline src/Accounts/Authentication/Factories/AuthenticationFactory.cs:192]: I rewrote the message so it's more aligned to the sceanrio. Please update the second string as well.
```suggestion
                        errorMessage = $"Failed to authenticate silently. User inteaction is required. This may be due to the access policy of multi-factor authentication to access tenant {tenantId}. Please rerun 'Connect-AzAccount' with additional parameter '-TenantId {tenantId}'.";
```

**Review by isra-fel — CHANGES_REQUESTED** [review]

File: src/Accounts/Authentication/Factories/AuthenticationFactory.cs (hunk 10 above)
Comment (isra-fel) [reply src/Accounts/Authentication/Factories/AuthenticationFactory.cs:192]: Also we should put the msal exception in debug log

**Review by isra-fel — APPROVED** [review]
